android/build.gradle

```
@@ -192,3 +192,58 @@ dependencies {
         }
     }
 }
+
+def isScreensExampleApp() {
+    return project.hasProperty('isScreensExampleApp') && project.property('isScreensExampleApp')
+}
+
+task copyCodegenArtifacts(type: Copy) {
+    group 'After build tasks'
+    description 'Tasks which copy codegen artifacts to paper architecture'
+    
+    if (!isScreensExampleApp()) {
```

And add check here for new architecture

```suggestion
    if (!isScreensExampleApp() || !isNewArchitectureEnabled()) {
```

---

Changed in https://github.com/software-mansion/react-native-screens/pull/2168/commits/bd65c0c7bb6d6a12facbf79e40c1a04b89aebb4f